modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -318,7 +318,7 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
          * @param[in] top Next layer to be fused.
          * @returns True if fusion was performed.
          */
-        virtual bool tryFuse(Ptr<Layer>& top);
+        virtual bool tryFuse(std::vector< Ptr<Layer> >& bottoms, Ptr<Layer>& top);
```

Comment:
I believe "layer" should know own inputs ("bottoms") - directly or via `Network`.